src/curses.rs

```diff
@@ -79,6 +79,13 @@
     pub clr_eol: Option<CString>,
     pub clr_eos: Option<CString>,
 
+    // OSC 133  docs: https://iterm2.com/documentation-escape-codes.html
+    pub prompt_start: Option<CString>,  // "\x1b133;A\x07"
+    pub prompt_end: Option<CString>,    // "\x1b133;B\x07"
+    pub command_start: Option<CString>, // "\x1b133;C\x07"
+    // pub command_end_before_status: Option<CString>, // "\x1b133;D;<exitcode>;\x07"
+    // pub command_end_after_status: Option<CString>,
```

Comment:
Might as well add command start/end too here.
Emit it next to where we emit `fish_preexec`/`fish_postexec`, passing `eval_res.status.status_value()`.
If you choose to do that, make sure it plays well with fish's default prompt which already renders a failing status in bold red.